internal/app/osd/internal/reg/reg.go

```diff
@@ -397,3 +400,66 @@ func (r *Registrator) Top(ctx context.Context, in *empty.Empty) (reply *proto.To
 	reply = &proto.TopReply{ProcessList: p}
 	return
 }
+
+// DF implements the proto.OSDServer interface.
+func (r *Registrator) DF(ctx context.Context, in *empty.Empty) (reply *proto.DFReply, err error) {
+	var stat unix.Statfs_t
+	file, err := os.Open("/proc/mounts")
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	stats := []*proto.DFStat{}
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		fields := strings.Fields(scanner.Text())
+
+		filesystem := fields[0]
+		mountpoint := fields[1]
+
+		f, err := os.Stat(mountpoint)
+		if err != nil {
+			continue
+		}
+		switch mode := f.Mode(); {
+		case !mode.IsDir():
+			continue
+		}
+
+		if err := unix.Statfs(mountpoint, &stat); err != nil {
+			continue
+		}
+
+		totalSize := uint64(stat.Bsize) * stat.Blocks
+		totalAvail := uint64(stat.Bsize) * stat.Bavail
+		totalUsed := totalSize - totalAvail
+		percentAvail := float64(totalAvail) / (float64(totalSize))
```

Comment:
nitpicking: extra parentheses in the 2nd part (around totalSize)

extra nitpicking: is it really `percent` if it's not multiplied by 100?